npchighlight/src/main/java/net/runelite/client/plugins/npchighlight/NpcSceneOverlay.java

```diff
@@ -218,18 +218,35 @@ private void renderNpcOverlay(Graphics2D graphics, NPC actor, Color color)
 			{
 				int size = 1;
 				NPCDefinition composition = actor.getTransformedDefinition();
-
 				if (composition != null)
 				{
 					size = composition.getSize();
 				}
-
-				final WorldPoint wp = actor.getWorldLocation();
-				final Color squareColor = color;
-
-				getSquare(wp, size).forEach(square ->
-					drawTile(graphics, square, squareColor, 1, 255, 50));
-				break;
+				LocalPoint lp = LocalPoint.fromWorld(client, actor.getWorldLocation());
+				if (lp != null)
+				{
+					lp = new LocalPoint(lp.getX() + size * 128 / 2 - 64, lp.getY() + size * 128 / 2 - 64);
+					final Polygon tilePoly = Perspective.getCanvasTileAreaPoly(client, lp, size);
+					renderPoly(graphics, color, tilePoly);
+					break;
```

Comment:
get the `break`'s out of the if statements. this way they fall through

Reply:
Done